src/utils/TextDecoderPolyfill.js

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2017 Vector Creations Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+"use strict";
+
+// Polyfill for TextDecoder.
+
+const REPLACEMENT_CHAR = '\uFFFD';
+
+export default class TextDecoder {
```

Comment:
I ended up writing it myself, because I couldn't find any other plausible implementations. As I wrote in the PR: 

> I'm thinking of factoring the TextEncoderPolyfill [and Decoder] out to a separate npm.js module - wdyt?